### ISSUE TITLE
MOM6 single column test crashes FATAL: input domain does not have an io_domain.

### ISSUE THREAD

**Issue by nikizadehgfdl** [NOAA-GFDL/FMS#497]
**Describe the bug**
There is a very old MOM6 single column test case that crashes  in FMS2020.02 like:
```
FATAL: input domain does not have an io_domain.
Image              PC                Routine            Line        Source             
fms_mom6_solo_com  0000000000EABEA1  mpp_mod_mp_mpp_er          69  mpp_util_mpi.inc
fms_mom6_solo_com  00000000012F2870  fms_io_utils_mod_         145  fms_io_utils.F90
fms_mom6_solo_com  000000000128C965  fms_netcdf_domain         343  fms_netcdf_domain_io.F90
fms_mom6_solo_com  00000000011607E1  data_override_mod        1271  data_override.F90
fms_mom6_solo_com  0000000001171C39  data_override_mod         452  data_override.F90
fms_mom6_solo_com  00000000006B88D3  mom_surface_forci         837  MOM_surface_forcing.F90
fms_mom6_solo_com  000000000069FAAA  mom_surface_forci         272  MOM_surface_forcing.F90
fms_mom6_solo_com  0000000000403705  MAIN__                    475  MOM_driver.F90
```
Here's line 837 of MOM_surface_forcing.F90
```
    call data_override_init(Ocean_domain_in=G%Domain%mpp_domain)
```

**To Reproduce**
1. Build mom6_solo executable 
fremake -x /ncrc/home2/Niki.Zadeh/xmls/xanadu/mdt_xml_RTS/MOM6_solo_postxanadu.xml -p ncrc4.intel18 -t repro mom6_solo_compile
2. cd mom6/ocean_only/single_column/EPBL
3. srun -n 1 fms_mom6_solo_compile.x  (on a compute node)

**Expected behavior**
Regression should not crash

**System Environment**
CLE7 / Intel18

**Additional context**
I doubt this is a CLE7 issue, rather it's an issue with FMS2020.02
Regression test passes with FMS2019.01.02

**Comment by wrongkindofdoctor** [comment]
@nikizadehgfdl If this experiment uses only on pe, you'll have to manually define the io domain because MOM does not set it up for single-pe jobs. I define the IO domain in create_file and domain-decomposed read/write routines as follows:
```
! define the io domain if on one pe and the io domain is not set
if (mpp_get_domain_npes(domain%mpp_domain) .eq. 1 ) then
  if (.not. associated(mpp_get_io_domain(domain%mpp_domain))) &
    call mpp_define_io_domain(domain%mpp_domain, (/1,1/))
endif
```

**Comment by nikizadehgfdl** [comment]
@adcroft , @marshallward  do you think the fix should be in FMS or MOM6 as suggested above?

**Comment by nikizadehgfdl** [comment]
@wrongkindofdoctor did you test this code for  this or a similar issue? Thanks anyway, will try.

**Comment by wrongkindofdoctor** [comment]
@marshallward @nikizadehgfdl @adcroft I ran into this problem during my new IO development when testing with some 1-pe experiments, and inserted the code in the MOM_io routines I modified to fix it since I was advised not to modify the code outside of the scope of the MOM_io and MOM_restart interfaces.
It would make more sense to get rid of the [logic block that only defines the io domain if the io_layout is not 0,0, and layout(1)*layout(2)>1](https://github.com/NOAA-GFDL/MOM6/blob/d44e79b9b16310d0539352ab48bdd6dd941d8b94/src/framework/MOM_domains.F90#L1592) in MOM_domains.

**Comment by adcroft** [comment]
Yes we agree. Although the current block is only for an optional diagnostic domain and not the model domain we do agree we should setup the io_domain systematically. The MOM6 change needs to be one that works across the 2019/2020 versions. It the change in behavior that caught us out here because we didn't need to do anything about the io_domain until 2020. I think will be solvable in MOM_domains.
